docs/localization/LOCALIZATION.md

```diff
@@ -74,343 +80,430 @@ export function localizationConfigurationFactory() {
+- In case it is not enough to provide your configuration object as explained above, you can also provide a new factory for the localization configuration.
+  You only need to provide a token `LOCALIZATION_CONFIGURATION_TOKEN` with your custom factory and dependencies.
 
 ```typescript
-// index.ts
-
-import { LOCALIZATION_CONFIGURATION_TOKEN, LocalizationModule, translateLoaderProvider } from  "@o3r/localization";
+// app.module.ts
 
-import {BidiModule} from '@angular/cdk/bidi';
-import { TranslateModule } from  "@ngx-translate/core";
-...
-import localeAR from  "@angular/common/locales/ar";
-import localeEN from  "@angular/common/locales/en";
-import localeFR from  "@angular/common/locales/fr";
-
-
-...
+import localeAR from '@angular/common/locales/ar';
+import localeEN from '@angular/common/locales/en';
+import localeFR from '@angular/common/locales/fr';
+import { TranslateModule } from '@ngx-translate/core';
+import { LOCALIZATION_CONFIGURATION_TOKEN, LocalizationModule, translateLoaderProvider } from '@o3r/localization';
+// ...
 
 registerLocaleData(localeEN, 'en-EN');
 registerLocaleData(localeFR, 'fr');
 registerLocaleData(localeAR, 'ar');
 
 @NgModule({
   imports: [
-  LocalizationModule,
-  BidiModule,
-  TranslateModule.forRoot({
-    loader: translateLoaderProvider,
-  }),
-  ...
+    LocalizationModule,
+    TranslateModule.forRoot({
+      loader: translateLoaderProvider,
+    })
+    // ...
   ],
-  ...
+  // ...
   providers: [
     {provide: LOCALIZATION_CONFIGURATION_TOKEN, useFactory: customCreateLocalizationConfiguration, deps: [YourServiceNeededByTheFactory]}
   ]
+})
+class AppModule {}
 ```
 
-You may also need to set up your http server to accept **CORS** and update **connect-src** property of **csp.json** in your application.
+## HTTP server
 
-```typescript
+You may also need to set up your http server to accept **CORS** and update the `connect-src` property of `csp.json` in your application.
 
+```json5
 // configs/csp.json
 
 {
+  "connect-src": "http://example.com"
+}
+```
 
-"connect-src": "http://example.com"
+## Localizing the components
 
-}
+Now we are ready to localize the components of the application.
 
+### How to generate localization files for a component
+
+You can directly generate a localized component with the following command:
+```shell
+ng g component ComponentName --use-localization
+```
+Or you can localize it later with:
+```shell
+ng g localization-to-component --path='path/to/the/component/class/file'
 ```
+> [!NOTE]
+> Later you can add a localization key to a component with the following command:
+> ```shell
+> ng g localization-key-to-component --path='path/to/the/component/class/file' --key='localizationKey' --value='defaultValue'
```

Comment:
```suggestion
> ng g localization-key-to-component --path='path/to/the/component/class/file' --key='localizationKey' --value='Default value for localizationKey'
```
So it's easier to understand that the key should not contain space but the value can